mittab/apps/tab/forms.py

```diff
@@ -39,90 +44,104 @@ def __init__(self, *args, **kwargs):
-#            self.fields['debaters'].initial = [d.pk for d in instance.debaters.all()]
+    debaters = forms.ModelMultipleChoiceField(queryset=Debater.objects.all(),
+                                              widget=FilteredSelectMultiple("Debaters",
+                                                                            is_stacked=False))
+
+    #    def __init__(self, *args, **kwargs):
+    #        super(TeamForm, self).__init__(*args, **kwargs)
+    #        if kwargs.has_key('instance'):
+    #            instance = kwargs['instance']
+    #            self.fields['debaters'].initial = [d.pk for d in instance.debaters.all()]
 
     def clean_debaters(self):
         data = self.cleaned_data['debaters']
-        if not( 1 <= len(data) <= 2) :
-            raise forms.ValidationError("You must select 1 or 2 debaters!") 
+        if not (1 <= len(data) <= 2):
+            raise forms.ValidationError("You must select 1 or 2 debaters!")
         return data
-    
+
     class Meta:
         model = Team
 
+
 class TeamEntryForm(forms.ModelForm):
     number_scratches = forms.IntegerField(label="How many initial scratches?", initial=0)
-    debaters = forms.ModelMultipleChoiceField(queryset=Debater.objects.filter(team__debaters__isnull=True), 
-                                              widget=FilteredSelectMultiple("Debaters", 
-                                              is_stacked=False))
+    debaters = forms.ModelMultipleChoiceField(queryset=Debater.objects.filter(team__debaters__isnull=True),
+                                              widget=FilteredSelectMultiple("Debaters",
+                                                                            is_stacked=False))
+
     def clean_debaters(self):
         data = self.cleaned_data['debaters']
-        if not( 1 <= len(data) <= 2) :
-            raise forms.ValidationError("You must select 1 or 2 debaters!") 
+        if not (1 <= len(data) <= 2):
+            raise forms.ValidationError("You must select 1 or 2 debaters!")
         return data
 
     class Meta:
         model = Team
-        
+
+
 class ScratchForm(forms.ModelForm):
     team = forms.ModelChoiceField(queryset=Team.objects.all())
     judge = forms.ModelChoiceField(queryset=Judge.objects.all())
     scratch_type = forms.ChoiceField(choices=Scratch.TYPE_CHOICES)
+
     class Meta:
         model = Scratch
-        
+
+
 class DebaterForm(forms.ModelForm):
     class Meta:
         model = Debater
-        
-        
+
+
 def validate_speaks(value):
-    if not (21.0 <= value <= 29.0 or value == 0):
-        raise ValidationError(u'%s is an entirely invalid speaker score, try again.' % value)
-    
-#TODO: Rewrite this, it is ugly as hell
+    lower_bound = 5   # defaults
+    upper_bound = 45
+    try:
+        lower_bound = int(TabSettings.objects.get(key="lower_speaks").value)
+        upper_bound = int(TabSettings.objects.get(key="upper_speaks").value)
+    except Exception as e:
```

Comment:
Also, when you use a try-except, make sure to specify the error class. Otherwise something like a typo in a function name could sleep by and we'd never notice